File: scripts/copy_frontend_vendor_assets.js
#!/usr/bin/env node
/**
 * Copy vendored frontend assets from node_modules into static/dist.
 * Do not use public CDNs.
 */
const fs = require("fs");
const path = require("path");

const root = path.resolve(__dirname, "..");
const htmxSrc = path.join(root, "node_modules", "htmx.org", "dist", "htmx.min.js");
const htmxDest = path.join(root, "static", "dist", "js", "htmx.min.js");
const appJsSrc = path.join(root, "static", "src", "js", "app.js");
const appJsDest = path.join(root, "static", "dist", "js", "app.js");

function mustCopy(src, dest) {
  if (!fs.existsSync(src)) {
    console.error(`Missing vendor asset: ${src}`);
    process.exit(1);
  }
  fs.mkdirSync(path.dirname(dest), { recursive: true });
  fs.copyFileSync(src, dest);
  console.log(`Copied ${path.relative(root, src)} -> ${path.relative(root, dest)}`);
}

const recordingJsSrc = path.join(root, "static", "src", "js", "recording-editor.js");
const recordingJsDest = path.join(root, "static", "dist", "js", "recording-editor.js");

mustCopy(htmxSrc, htmxDest);
mustCopy(appJsSrc, appJsDest);
mustCopy(recordingJsSrc, recordingJsDest);

File: maintainpro/systems/fg-digital-recording/scripts/copy_frontend_vendor_assets.py
#!/usr/bin/env python3
"""Copy vendored frontend assets from node_modules into static/dist (no CDN)."""

from __future__ import annotations

import shutil
import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]


def must_copy(src: Path, dest: Path) -> None:
    if not src.exists():
        print(f"Missing vendor asset: {src}", file=sys.stderr)
        raise SystemExit(1)
    dest.parent.mkdir(parents=True, exist_ok=True)
    shutil.copy2(src, dest)
    print(f"Copied {src.relative_to(ROOT)} -> {dest.relative_to(ROOT)}")


def main() -> int:
    must_copy(
        ROOT / "node_modules" / "htmx.org" / "dist" / "htmx.min.js",
        ROOT / "static" / "dist" / "js" / "htmx.min.js",
    )
    must_copy(
        ROOT / "static" / "src" / "js" / "app.js",
        ROOT / "static" / "dist" / "js" / "app.js",
    )
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
